chain/gen/gen.go

```diff
@@ -476,8 +476,18 @@ func (cg *ChainGen) NextTipSetFromMinersWithMessagesAndNulls(base *types.TipSet,
 	}
 
 	fts := store.NewFullTipSet(blks)
+	if err := cg.cs.PersistTipsets(context.TODO(), []*types.TipSet{fts.TipSet()}); err != nil {
+		return nil, xerrors.Errorf("failed to persist tipset: %w", err)
+	}
+
+	for _, blk := range fts.TipSet().Blocks() {
```

Comment:
Not strictly necessary since we do this in `PutTipSet` too, but doesn't seem harmful to do for tests.